tests/dev-utils/deltareplayer.js

```diff
@@ -149,6 +131,26 @@ describe( 'DeltaReplayer', () => {
 
 			return deltaReplayer.play();
 		} );
+
+		it( 'should correctly handle errors coming from the engine', () => {
+			const doc = getDocument();
+
+			const invalidDelta = getSecondDelta();
+			invalidDelta.operations[ 0 ].baseVersion = 3;
+
+			const stringifiedDeltas = [ getFirstDelta(), invalidDelta ]
+				.map( d => JSON.stringify( d ) )
+				.join( '---' );
+
+			const deltaReplayer = new DeltaReplayer( doc, '---', stringifiedDeltas );
+
+			return deltaReplayer.play( 1 )
+				.then( () => {
+					throw new Error( 'It should throw an error' );
+				}, ( err ) => {
+					expect( err.name ).to.equal( 'CKEditorError' );
```

Comment:
It's worth checking the message. We usually check it with `expect( err.message ).to.match( /^the-error-identifier:/ )` – this is precise enough, but you're also sure that you caught the right error. Not any CKEditorError.